File: 1110.py
#https://www.acmicpc.net/problem/1110
#더하기 사이클
'''
#방법1
N = int(input())
result = N
count = 0

while True:
    first = result//10
    second = result%10
    sumRight = (first + second)%10
    result = (second*10) + sumRight
    count+=1
    if N == result:
        break
print(count)
'''

#방법2
N = int(input())
original = N
cnt = 0

while True:
    cnt+=1
    try:
        first = N // 10
        second = N % 10
        N = first + second
        if N >= 10:
            N = N % 10
        N = second * 10 + N
        if N == original:
            break;
    except:
        N = 10*N + N
print(cnt)

File: js-baekjoon/1110.js
//https://www.acmicpc.net/problem/1110
//더하기 사이클

const readline = require('readline')
const r1 = readline.createInterface({
    input : process.stdin,
    output : process.stdout
})

r1.on('line', (line) => {
    input = Number(line)
    let num = input
    let sum
    let cnt = 0

    while(true){
        cnt++
        sum = Math.floor(num / 10) + num % 10;
        if(sum > 9){
            sum = sum % 10
        }

        num = (num % 10)*10 + sum

        if (input === num) {
            break
        }
    }
    console.log(cnt)
    r1.close()
}).on('close', () =>{
    process.exit()
})